libs/s25main/LeatherLoader.h

```diff
@@ -17,60 +17,71 @@ class GameWorldBase;
 
 namespace leatheraddon {
 
-bool isLeatherAddonBuildingType(BuildingType bld);
-bool isLeatherAddonGoodType(GoodType good);
-bool isLeatherAddonJobType(Job job);
+inline bool isLeatherAddonBuildingType(BuildingType bld)
+{
+    return bld == BuildingType::Skinner || bld == BuildingType::Tannery || bld == BuildingType::LeatherWorks;
+}
+
+inline bool isLeatherAddonGoodType(GoodType good)
+{
+    return good == GoodType::Skins || good == GoodType::Leather || good == GoodType::Armor;
+}
+
+inline bool isLeatherAddonJobType(Job job)
+{
+    return job == Job::Skinner || job == Job::Tanner || job == Job::LeatherWorker;
+}
 
-enum class BobTypes
+enum class BobType
 {
-    INVALID,
-    SKINNER_SKINNING_ANIMAL_CARCASS_ANIMATION,
-    SKINNER_WALKING,
-    SKINNER_CARRYING_SKINS,
-    TANNERY_WORK_WINDOW_ANIMATION,
-    TANNER_WALKING,
-    TANNER_CARRYING_LEATHER_IN_OUT,
-    LEATHERWORKS_WORK_WINDOW_ANIMATION,
-    LEATHERWORKER_WALKING,
-    LEATHERWORKER_CARRYING_ARMOR_IN_OUT,
-    THIN_CARRIER_CARRYING_SKINS,
-    FAT_CARRIER_CARRYING_SKINS,
-    THIN_CARRIER_CARRYING_LEATHER,
-    FAT_CARRIER_CARRYING_LEATHER,
-    THIN_CARRIER_CARRYING_ARMOR,
-    FAT_CARRIER_CARRYING_ARMOR,
-    DISTRIBUTION_OF_PIGS_ICON,
-    SKINS_WARE_ICON,
-    LEATHER_WARE_ICON,
-    LEATHERWORKING_WARES_TRANSPORT_PRIORITY_TREE_ICON,
-    ARMOR_WARE_ICON,
-    ARMOR_DELIVER_ICON,
-    DISABLE_DELIVERY_ARMOR_ICON,
-    SKINNER_JOB_ICON,
-    TANNER_JOB_ICON,
-    LEATHERWORKER_JOB_ICON,
-    SKINS_WARE_ON_GROUND_OF_FLAG,
-    DONKEY_BOAT_CARRYING_SKINS_WARE,
-    LEATHER_WARE_ON_GROUND_OF_FLAG,
-    DONKEY_BOAT_CARRYING_LEATHER_WARE,
-    ARMOR_WARE_ON_GROUND_OF_FLAG,
-    DONKEY_BOAT_CARRYING_ARMOR_WARE,
-    STOP_COINS_X_SIGN_OVERRIDE,
-    STOP_ARMOR_X_SIGN,
+    Invalid,
```

Comment:
Do you need this?  
It was historically used in some enums before "std::optional" and "OptionalEnum" existed.

Here it seems it is only to designate a situation that should never appear: returned from a function and then asserts that it isn't returned. That sounds like a precondition violation of the function so it could e.g. throw when detected.